src/rgw/rgw_rados.h

```diff
@@ -395,8 +396,172 @@ template <class K, class V>
 class lru_map;
 using tombstone_cache_t = lru_map<rgw_obj, tombstone_entry>;
 
+struct get_obj_io {
+  off_t len;
+  bufferlist bl;
+};
+
+
+struct get_obj_aio_data {
+  struct get_obj_data *op_data;
+  off_t ofs;
+  off_t len;
+};
+
 class RGWIndexCompletionManager;
 
+class librados::CacheRequest {
+  public:
```

Comment:
I feel like I have commented on this a number of times, this does not belong in librados namespace

Reply:
moved out of librados namespace to 
https://github.com/ceph/ceph/blob/220acce1b33a677c6a4a9609f15cc974ba4242a7/src/rgw/rgw_cacherequest.h